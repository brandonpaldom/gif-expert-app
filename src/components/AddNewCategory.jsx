import { useState } from 'react'
import PropTypes from 'prop-types'

export function AddNewCategory({ handleAddNewCategory }) {
  const [categoryInputValue, setCategoryInputValue] = useState('')

  function handleCategoryInputChange({ target }) {
    setCategoryInputValue(target.value)
  }

  function handleCategorySubmit(e) {
    e.preventDefault()
    if (categoryInputValue.trim().length <= 1) return
    handleAddNewCategory(categoryInputValue.trim())
    setCategoryInputValue('')
  }

  return (
    <form
      onSubmit={handleCategorySubmit}
      className="flex flex-col"
      aria-label="form"
    >
      <input
        type="text"
        placeholder="Search for gifs..."
        value={categoryInputValue}
        onChange={handleCategoryInputChange}
        className="border-none bg-neutral-900 px-4 py-3 transition-colors duration-300 placeholder:text-neutral-600 focus:border-none focus:bg-neutral-800 focus:outline-none focus:ring-0"
      />
    </form>
  )
}

AddNewCategory.propTypes = {
  handleAddNewCategory: PropTypes.func.isRequired,
}
