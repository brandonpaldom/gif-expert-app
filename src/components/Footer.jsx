export function Footer() {
  return (
    <header className="flex h-16 items-center bg-black px-4">
      <p className="text-[0.875rem] text-white">
        Creado por <span className="text-green-400">Brandon Palmeros</span>
      </p>
    </header>
  );
}
