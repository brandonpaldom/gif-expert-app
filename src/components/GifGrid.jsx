import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifItem, GifLoader } from './'
import PropTypes from 'prop-types'

export function GifGrid({ gifCategory }) {
  const { images, isLoading } = useFetchGifs(gifCategory)

  if (isLoading) return <GifLoader />

  return (
    <div className="flex flex-col gap-4">
      <h2 className="text-[2rem] font-bold capitalize text-green-400">
        {gifCategory}
      </h2>
      <ul className="grid grid-cols-2 gap-4 md:grid-cols-3 lg:grid-cols-4">
        {images.map((image) => (
          <GifItem key={image.gifId} {...image} />
        ))}
      </ul>
    </div>
  )
}

GifGrid.propTypes = {
  gifCategory: PropTypes.string.isRequired,
}
