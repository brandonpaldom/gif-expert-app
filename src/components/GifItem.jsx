import PropTypes from 'prop-types'

export function GifItem({ gifTitle, gifUrl }) {
  return (
    <li className="flex flex-col gap-2">
      <img src={gifUrl} alt={gifTitle} />
      <p className="text-[0.75rem] text-neutral-600">{gifTitle}</p>
    </li>
  )
}

GifItem.propTypes = {
  gifTitle: PropTypes.string.isRequired,
  gifUrl: PropTypes.string.isRequired,
}
