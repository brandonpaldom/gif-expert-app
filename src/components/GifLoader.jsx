export function GifLoader() {
  const loaders = [1, 2, 3, 4, 5, 6, 7, 8];

  return (
    <div className="flex flex-col gap-4">
      <div className="h-8 w-2/3 animate-pulse rounded-full bg-neutral-900 md:w-1/4 lg:w-1/3" />
      <div className="grid grid-cols-2 gap-4 md:grid-cols-3 lg:grid-cols-4">
        {loaders.map((loader) => (
          <div
            key={loader}
            className="aspect-square animate-pulse bg-neutral-900"
          />
        ))}
      </div>
    </div>
  );
}
