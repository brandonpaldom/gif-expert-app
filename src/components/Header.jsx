import giphyLogo from '../assets/giphy-logo.png';

export function Header() {
  return (
    <header className="flex h-16 items-center bg-black px-4">
      <img src={giphyLogo} alt="" />
    </header>
  );
}
