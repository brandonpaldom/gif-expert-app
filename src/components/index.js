export * from './AddNewCategory';
export * from './Footer';
export * from './GifGrid';
export * from './GifItem';
export * from './GifLoader';
export * from './Header';
