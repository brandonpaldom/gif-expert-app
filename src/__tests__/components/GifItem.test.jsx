import { render, screen } from '@testing-library/react'
import { GifItem } from '../../components'

describe('GifItem', () => {
  const title = 'Gif title'
  const url = 'https://media.giphy.com/media/3o7TKsQ8gqVrXQrjWM/giphy.gif'

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifItem gifTitle={title} gifUrl={url} />)
    expect(container).toMatchSnapshot()
  })

  test('Debe mostrar el título y la imagen', () => {
    render(<GifItem gifTitle={title} gifUrl={url} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('Debe mostrar el título', () => {
    render(<GifItem gifTitle={title} gifUrl={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
