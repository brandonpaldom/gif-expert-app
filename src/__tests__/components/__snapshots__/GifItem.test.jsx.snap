// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-2"
  >
    <img
      alt="Gif title"
      src="https://media.giphy.com/media/3o7TKsQ8gqVrXQrjWM/giphy.gif"
    />
    <p
      class="text-[0.75rem] text-neutral-600"
    >
      Gif title
    </p>
  </div>
</div>
`;
